framework/doc/content/source/csg/CSGBase.md

```diff
@@ -0,0 +1,663 @@
+There are 3 different behaviors for joining bases that are supported depending on the additional arguments that are passed:
+
+1. No additional arguments: All cells that are in the root universe of the incoming `CSGBase` object will be added to the existing root universe of the current base object, and the root universe from the incoming base will no longer exist.
+
+```cpp
+// get a list of bases associated with a list of mesh generator names
+const auto csg_bases = getCSGBases("input_mg");
+// first mesh generator base will be used as the main and all others will be joined to this one.
+std::unique_ptr<CSG::CSGBase> csg_obj = std::move(*csg_bases[0]);
+for (unsigned int i = 1; i < csg_bases.size(); ++i)
+{
+  // get the specific base object from the input and join to the first
+  std::unique_ptr<CSG::CSGBase> inp_csg_obj = std::move(*csg_bases[i]);
+  // all cells in root for inp_csg_obj are moved to the root of the current csg_obj
+  csg_obj->joinOtherBase(inp_csg_obj);
+}
+```
+
+2. One new root universe name (`new_root_name_join`): All cells in the root universe of the incoming base will be used to create a new universe of the name specified by the `new_root_name_join` parameter. These cells will *not* be added to the existing root universe. This new universe will be added as a new non-root universe in the existing base object. *This newly created universe will not be connected to the root universe of the existing `CSGBase` object by default.*
+
+```cpp
+// get a list of bases associated with a list of mesh generator names
+const auto csg_bases = getCSGBasesByName(input_mg_names);
+// first mesh generator base will be used as the main
+// all others will be joined to this one creating a new universe based on the root of the incoming base.
+std::unique_ptr<CSG::CSGBase> csg_obj = std::move(*csg_bases[0]);
+for (unsigned int i = 1; i < input_mg_names.size(); ++i)
+{
+  // get the CSGBase obj for the associated incoming MG
+  inp_csg_obj = std::move(*csg_bases[i]);
+  // specify a new name for the universe to be created from the incoming root universe
+  new_join_name = input_mg_names[i] + "_univ";
+  // the root universe of csg_obj will remain unchanged
+  // the root universe of inp_csg_obj will be renamed and maintained as a new separate universe
+  csg_obj->joinOtherBase(inp_csg_obj, new_join_name);
+}
+```
+
+3. Two new root universe names (`new_root_name_base` and `new_root_name_join`): The cells in the root universe of the current `CSGBase` object will be used to create a new non-root universe of the name specified by the `new_root_name_base` parameter, and the cells in the root universe of the incoming `CSGBase` object will be used to create a separate non-root universe of the name specified by the `new_root_name_join` parameter. *At the end of this join method, the root universe of the current base object will be empty and neither of the two new non-root universes will be connected to the root universe by default.*
+
```

Comment:
I think it would help to explain a brief use case for this?

Reply:
I think it would be hard to come up with a "brief" use case here that wouldn't require images or a lot of background. As of now, we don't have an exact place in mind where this method would be used, but I think it is feasible that the flexibility for joining in this way will eventually be needed. When we start implementing this in MGs, if we end up using it, we could link to that MG as an example use.